src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -112,9 +112,19 @@ public int Count
         
         public override int GetHashCode()
         {
-            return null == _array
-                        ? 0
-                        : _array.GetHashCode() ^ _offset ^ _count;
+            if (_array == null)
+            {
+                return 0;
+            }
+            
+            int hash = 5381;
+            hash = System.Numerics.Hashing.HashHelpers.Combine(hash, _offset);
```

Comment:
Unfortunately, the fully-qualified name has to be used since this clashes with `System.Collections.HashHelpers`.